File: problems/67/solution.cs
public class Solution {
    public string AddBinary(string a, string b) {
        var an = a.Length - 1;
        var bn = b.Length - 1;
        var carry = false;
        var res = "";
        
        while (an >= 0 || bn >= 0) {
            var nch = carry ? 1 : 0;
            nch += an >= 0 && a[an] == '1' ? 1 : 0;
            nch += bn >= 0 && b[bn] == '1' ? 1 : 0;
            an--;
            bn--;
            carry = nch >= 2;
            nch -= carry ? 2 : 0;
            
            res = nch.ToString() + res;
        }
        
        if (carry) {
            res = "1" + res;
        }
        
        return res;
    }
}

File: problems/67/solution.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        int an = a.size() - 1, bn = b.size() - 1;
        string res = "";
        bool carry = false;
        
        while (an >= 0 || bn >= 0 || carry) {
            int nch = carry ? 1 : 0;
            nch += an >= 0 && a[an] == '1' ? 1 : 0;
            nch += bn >= 0 && b[bn] == '1' ? 1 : 0;
            an--;
            bn--;
            carry = nch >= 2;
            nch %= 2;
            res = to_string(nch) + res;
        }
        
        return res;
    }
};